### ISSUE TITLE
syntax for inserting multiple sub-objects

### ISSUE THREAD

**Issue by will123195** [SkyPHP/skyphp#8]
given this aql:

```
news_item {
    message,
    news_who {
        who
    }
}
```

``` php
<?
// this works
        $n = new news_item();
        $n->message = $html_message;
        foreach ( $who_array as $i => $who ) {
            $n->news_who[$i]['who'] = $who;
        }
        $n->save();

// this does not work
        $n = new news_item();
        $n->message = $html_message;
        foreach ( $who_array as $i => $who ) {
            $n->news_who[$i]->who = $who;
        }
        $n->save();
```

**Comment by will123195** [comment]
stan, i thought you said you fixed this, but it's not fixed so i'm leaving this issue open

**Comment by stanistan** [comment]
``` php
<?

$n->news_who[$i]->who = $who;

```

This works when `$n->news_who[$i]` is already set.

However it otherwise automatically casts this as stdClass which I never tested for, only arrays, arrayObjects, and model objects.

In this case, the fix is that it will now work on `$n->save()`, and `$n->dataToArray()`, but:

``` php
<?

($n->news_who[$i]['who'] == $n->news_who[$i]->who); // false because of how it was set originally

// other case

$n->news_who[$i] = array();
$n->news_who[$i]->who = $whatever;

($n->news_who[$i]['who'] == $n->news_who[$i]->who); // will be true

```

**Comment by will123195** [comment]
resolved in 3.0
